pkg/network/namescheme/networknamescheme_test.go

```diff
@@ -20,11 +20,75 @@
 package namescheme_test
 
 import (
-	"testing"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
 
-	"kubevirt.io/client-go/testutils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
 )
 
-func TestPodInterfaceNameScheme(t *testing.T) {
-	testutils.KubeVirtTestSuiteSetup(t)
+var _ = Describe("Network Name Scheme", func() {
+	Context("CreateNetworkNameScheme", func() {
+		vmiWithNetworks := func(networkList []virtv1.Network) *virtv1.VirtualMachineInstance {
+			return &virtv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-vmi",
+				},
+				Spec: virtv1.VirtualMachineInstanceSpec{
+					Networks: networkList,
+				},
+			}
+		}
+		DescribeTable("should return the expected NetworkNameSchemeMap",
+			func(networkList []virtv1.Network, expectedNetworkNameSchemeMap map[string]string) {
+				vmi := vmiWithNetworks(networkList)
+				podIfacesNameScheme := namescheme.CreateNetworkNameScheme(vmi)
+
+				Expect(podIfacesNameScheme).To(Equal(expectedNetworkNameSchemeMap))
+			},
+			Entry("when network list is nil", nil, map[string]string{}),
+			Entry("when no multus networks exist",
+				[]virtv1.Network{
+					newPodNetwork("default"),
+				},
+				map[string]string{
+					"default": namescheme.PrimaryPodInterfaceName,
+				}),
+			Entry("when default multus networks exist",
+				[]virtv1.Network{
+					newMultusNetwork("network0", "default/nad0", true),
+					newMultusNetwork("network1", "default/nad1", false),
+					newMultusNetwork("network2", "default/nad2", false),
+				},
+				map[string]string{
+					"network0": namescheme.PrimaryPodInterfaceName,
+					"network1": "net1",
+					"network2": "net2",
+				}),
+		)
+	})
+})
+
+func newMultusNetwork(name, networkName string, isDefault bool) virtv1.Network {
```

Comment:
Could you please apply the same pattern from previous commit here? (avoid passing a boolean to function)

Reply:
Done